azure/client.go

```diff
@@ -58,12 +58,15 @@ func (c client) ListAppSecurityGroups(rgName string) ([]string, error) {
 	ctx := context.Background()
 	groups := []string{}
 
-	for list, err := c.sgClient.ListComplete(ctx, rgName); list.NotDone(); err = list.Next() {
+	pager := c.sgClient.NewListAllPager(nil)
+	for pager.More() {
+		nextResult, err := pager.NextPage(ctx)
 		if err != nil {
-			return groups, fmt.Errorf("List Complete App Security Groups: %s", err)
+			return groups, fmt.Errorf("failed to advance page: %s", err)
```

Comment:
Same with the return values here: empty list and original err wrap msg for consistency.

Reply:
done